examples/DQN_variant/train.py

```diff
@@ -0,0 +1,196 @@
+        frame_skip=4,
+        context_len=4)
+
+    env.seed(config['train_env_seed'])
+    test_env.seed(config['test_env_seed'])
+
+    act_dim = env.action_space.n
+    config['act_dim'] = act_dim
+
+    logger.set_dir(f'./train_log/{algo_name}/{curr_time}')
+    logger.info('env {}, train obs_dim {}, train act_dim {}'.format(config['env_name'], env.observation_space.shape, act_dim))
+    logger.info(f'current configs are : \n {config}')
+
+    rpm = ReplayMemory(config['memory_size'], (dim, dim), 4)
+
+    # build an agent
+    model = AtariModel(act_dim=act_dim, dueling=config['dueling'])
+
+    if algo_name == 'DQN':
+        alg = DQN(model, gamma=config['gamma'], lr=config['lr_start'])
+
+    elif algo_name == 'DDQN':
+        alg = DDQN(model, gamma=config['gamma'], lr=config['lr_start'])
+
+    else:
+        pass
+
+    agent = AtariAgent(alg, config)
+
+    with tqdm(
+            total=config['memory_warmup_size'], desc='[Replay Memory Warm Up]') as pbar:
+        while rpm.size() < config['memory_warmup_size']:
+            total_reward, steps, _ = run_train_episode(agent, env, rpm)
+            pbar.update(steps)
+
+    test_flag = 0
+    train_total_steps = config['train_total_steps']
+    pbar = tqdm(total=train_total_steps)
+    cum_steps = 0
+
+    while cum_steps < train_total_steps:
+        
+        # start epoch
+        total_reward, steps, loss = run_train_episode(agent, env, rpm)
+        cum_steps += steps
+        pbar.set_description('[train]exploration:{}, learning_rate {}'.format(agent.curr_ep, alg.optimizer.get_lr()))
+        summary.add_scalar(f'{algo_name}/training_rewards', total_reward, cum_steps)
+        summary.add_scalar(f'{algo_name}/loss', loss, cum_steps)  # mean of total loss
+        summary.add_scalar(f'{algo_name}/exploration', agent.curr_ep, cum_steps)
+        summary.add_scalar(f'{algo_name}/learning_rate', alg.optimizer.get_lr(), cum_steps)
+
+        pbar.update(steps)
+
+        if cum_steps // config['eval_every_steps'] >= test_flag:
+
+            while cum_steps // config['eval_every_steps'] >= test_flag:
+                test_flag += 1
+
+            pbar.write("testing")
+
+            eval_rewards_mean, _ = run_evaluate_episodes(agent, test_env)
+
+            logger.info(
+                "eval_agent done, (steps, eval_reward): ({}, {})".format(
+                    cum_steps, eval_rewards_mean))
+
+            summary.add_scalar(f"{algo_name}/mean_{config['eval_episodes']}_validation_rewards", eval_rewards_mean, cum_steps)
+
+    pbar.close()
+
+    # final test score
+    eval_rewards_mean, eval_rewards = run_evaluate_episodes(agent, test_env, test=True)
+    std = np.std(eval_rewards)
+    logger.info(f"final mean {config['test_episodes']} test rewards is {eval_rewards_mean} +- {std}")
+
+    # save the parameters to ./model.ckpt
+    save_path = f'./model/{algo_name}/{curr_time}_model.ckpt'
+    agent.save(save_path)
+
+
+if __name__ == '__main__':
+    main()
```

Comment:
Please keep this file(train.py) basically the same as the fluid example(benchmark/fluid/DQN_varient/train.py)

Reply:
and please add README.md for this example.